### PR TITLE
[WIP] Global dictmerger

### PR DESCRIPTION
**Pull request by deepakn94**
Still need to make the updates atomic, but comments welcome!

### CODE REVIEW

File: tests/integration_tests.rs [hunk 1]
```diff
@@ -900,6 +900,60 @@ fn simple_for_dictmerger_loop() {
     unsafe { weld_value_free(ret_value) };
 }
 
+fn simple_for_global_dictmerger_loop() {
```
Comment (jjthomas) [inline tests/integration_tests.rs]: Is this function just a copy of `simple_for_dictmerger_loop` with the `global` annotation? If so, we should try to share the code between the two tests.

File: weld/resources/dictmerger_global.ll [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  %structSize = ptrtoint %$NAME* %structSizePtr to i64
+  %rawPtr = call i8* @new_merger(i64 %structSize, i32 1)
+  %newDict = call %$NAME @$NAME.new(i64 %capacity)
+  %bldPtr = bitcast i8* %rawPtr to %$NAME*
+  store %$NAME %newDict, %$NAME* %bldPtr, align 1
+  ret %$NAME.bld %bldPtr
+}
+
+; Append a value into a builder, growing its space if needed.
+define %$NAME.bld @$NAME.bld.merge(%$NAME.bld %bldPtr, %$KV_STRUCT %keyValue, i32 %workerId) {
```
Comment (jjthomas) [inline weld/resources/dictmerger_global.ll]: Can we share this code with the local `dictmerger`?

Comment (deepakn94) [inline weld/resources/dictmerger_global.ll]: Are you suggesting refactoring the linear probing logic to some sort of `dictmerger_prelude`?

Comment (jjthomas) [inline weld/resources/dictmerger_global.ll]: Right, the `dictmerger` struct is the same in both cases, right?

**Pull request closed without merging**